core/evm.go

```diff
@@ -30,7 +30,10 @@ import (
 )
 
 // NewEVMBlockContext creates a new context for use in the EVM.
-func NewEVMBlockContext(header *types.Header, blockHashFunc func(n uint64) common.Hash, engine consensus.Engine, author *common.Address) vm.BlockContext {
+//
+// excessDataGas must be set to the excessDataGas value from the *parent* block header, and can be
+// nil if the parent block is not of EIP-4844 type. It is read only.
+func NewEVMBlockContext(header *types.Header, excessDataGas *big.Int, blockHashFunc func(n uint64) common.Hash, engine consensus.Engine, author *common.Address) vm.BlockContext {
```

Comment:
Maybe I missed something, but seems like we can get `excessDataGas` from `header` with below, am I missing something?

Same question for ApplyTransaction

```
        header := block.Header()
	parentHeader := getHeader(header.ParentHash, header.Number.Uint64())
	var excessDataGas *big.Int
	if parentHeader != nil {
		excessDataGas = parentHeader.ExcessDataGas
	}
```

Reply:
there is no "getHeader" func available here though.